### PR TITLE
Moving Windows RunAsUserName feature to GA

### PR DESCRIPTION
**Pull request by marosset**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespace from that line:

/kind api-change

**What this PR does / why we need it**:
This PR moves Windows RunAsUserName kubelet feature to GA

**Which issue(s) this PR fixes**:

[Enhancement]: RunAsUserName for Windows kubernetes/enhancements#1043

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
Moving Windows RunAsUserName feature to GA
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

**Comment by marosset** [comment]
I'll update the failing/remove the tests which check that RunAsUserName gets dropped in API calls when the feature is disabled.

**Comment by PatrickLang** [comment]
wow dropped some bits there coming back from PTO. not milestone 1.14
/milestone v1.18

File: pkg/features/kube_features.go [hunk 1]
```diff
@@ -598,7 +599,7 @@ var defaultKubernetesFeatureGates = map[featuregate.Feature]featuregate.FeatureS
 	TTLAfterFinished:               {Default: false, PreRelease: featuregate.Alpha},
 	KubeletPodResources:            {Default: true, PreRelease: featuregate.Beta},
 	WindowsGMSA:                    {Default: true, PreRelease: featuregate.Beta},
-	WindowsRunAsUserName:           {Default: true, PreRelease: featuregate.Beta},
+	WindowsRunAsUserName:           {Default: true, PreRelease: featuregate.GA, LockToDefault: true},
```
Comment (ddebroy) [inline pkg/features/kube_features.go]: Should the `LockToDefault` be accompanied with a comment on which k/k version will be suitable to remove the flag completely? E.g. in https://github.com/kubernetes/kubernetes/pull/86210/files

**Review by ddebroy — COMMENTED** [review]
Please squash the commits. LGTM overall with one comment.

File: pkg/features/kube_features.go (hunk 1 above)
Comment (marosset) [reply pkg/features/kube_features.go]: I see a few features that have LockToDefault: true and no comment but i'll go ahead and add one for 1.20 anyways.
I'll rebase oncet hose changes are done.

**Comment by liggitt** [comment]
/approve

will defer to sig-windows reviews for final lgtm

**Review by michmike — APPROVED** [review]
/lgtm
/approve

**Pull request merged**